federate-clusters/README.adoc

```diff
@@ -0,0 +1,172 @@
+    tar -xzvf kubernetes-client-linux-amd64.tar.gz
+    sudo cp kubernetes/client/bin/kubefed /usr/local/bin
+    sudo chmod +x /usr/local/bin/kubefed
+    sudo cp kubernetes/client/bin/kubectl /usr/local/bin
+    sudo chmod +x /usr/local/bin/kubectl
+    # check context settings
+    kubectl config get-contexts
+    CURRENT   NAME                 CLUSTER              AUTHINFO             NAMESPACE
+              earth-k8s.internal   earth-k8s.internal   earth-k8s.internal
+              mars                 mars-k8s.internal    mars-k8s.internal
+    *         mars-k8s.internal    mars-k8s.internal    mars-k8s.internal
+              earth                earth-k8s.internal   earth-k8s.internal
+    # Use earth as host cluster
+    kubectl config use-context earth
+    # Create RBAC role binding
+    kubectl create clusterrolebinding admin-to-cluster-admin-binding --clusterrole=cluster-admin --user=admin
+
+. Deploy the federation control plane in host cluster
+
+    kubefed init interstellar --host-cluster-context=earth --dns-provider=aws-route53 --dns-zone-name=earth-mars-k8s.internal
+    Creating a namespace federation-system for federation system components... done
+    Creating federation control plane service..... done
+    Creating federation control plane objects (credentials, persistent volume claim)... done
+    Creating federation component deployments... done
+    Updating kubeconfig... done
+    Waiting for federation control plane to come up............................................................................ done
+    Federation API server is running at: a3c168c58981911e7b3e90af787d6efe-674228795.us-east-1.elb.amazonaws.com
+    # you can check newly created context
+    kubectl config get-contexts
+    # change the context to federation
+    kubectl config use-context interstellar
+    # create default namespace if you don't see it (possible bug)
+    kubectl get namespace --context=interstellar
+    kubectl create namespace default --context=interstellar
+
+. Join Earth and Mars cluster to the federation
+
+    kubefed join earth --host-cluster-context=earth --cluster-context=earth
+    kubefed join mars --host-cluster-context=earth --cluster-context=mars
+    # check status of cluster
+    # it took several hours for both clusters to show as Ready
+    kubectl --context=interstellar get clusters
+    NAME      STATUS    AGE
+    earth     Ready     13h
+    mars      Ready     13h
+
+. Deploy application
+
+    # deploy an application using replica sets
+    kubectl run nginx --image=nginx --replicas=2
+    kubectl get deployments -o wide
+    # didn't get successful deployments
+    # works for independent clusters but not on federation
+
+. Deploy Service
+
+    # expose the application using ALB
+    kubectl expose deployment nginx --type=LoadBalancer --port=80
+    kubectl get services -o wide
```

Comment:
can you add expected output?

Reply:
Shouldn't this be "ELB" and not "ALB", I am assuming the ALB controller is not installed here?

Reply:
+1 ALB ingress controller is another section

Reply:
you are correct. i will make the change